modules/order/src/Plugin/Commerce/Condition/OrderEmailAddress.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\commerce_order\Plugin\Commerce\Condition;
+
+use Drupal\commerce\Plugin\Commerce\Condition\ConditionBase;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides the Email address condition for orders.
+ *
+ * @CommerceCondition(
+ *   id = "order_email_address",
+ *   label = @Translation("Email address"),
+ *   display_label = @Translation("Limit by email address"),
+ *   category = @Translation("Customer"),
+ *   entity_type = "commerce_order",
+ * )
+ */
+class OrderEmailAddress extends ConditionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'mail' => NULL,
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $form['email'] = [
```

Comment:
You are using email and mail, instead of choosing one. That doesn't make sense. Let's just use "mail"